user_connections.go

```diff
@@ -0,0 +1,208 @@
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	handler, ok := m.handlers[nick]
+	return handler, ok
+}
+
+func (m *ConnectedUsersPool) AddConnection(conn *SockChatWS, store SockchatChannelStore) SockchatUserHandler {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		handler = NewUserHandler(conn, store)
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		m.handlers[conn.Nick] = handler
+	}
+	handler.AddConnection(conn)
+	return handler
+}
+
+func (m *ConnectedUsersPool) RemoveConnection(conn *SockChatWS) {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		log.Printf("could not drop connection - no handler found for %s", conn.Nick)
+		return
+	}
+
+	if handler.getActiveConnectionsCount() > 1 {
+		handler.RemoveConnection(conn)
+	} else {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		delete(m.handlers, conn.Nick)
+	}
+}
+
+// UserHandler manages connections of a single connected user
+type UserHandler struct {
+	Nick                  string
+	connections           map[*SockChatWS]bool
+	createChannelRequests chan *CreateChannel
+	joinChannelRequests   chan *JoinChannel
+	leaveChannelRequests  chan *LeaveChannel
+	sendMessageRequests   chan *SendMessage
+	lock                  sync.RWMutex
+	channelStore          SockchatChannelStore
+}
+
+type CreateChannel struct {
+	name        string
+	errCallback chan error
+}
+
+type JoinChannel struct {
+	channel     string
+	errCallback chan error
+}
+
+type LeaveChannel struct {
+	channel     string
+	errCallback chan error
+}
+
+type SendMessage struct {
+	channel     string
+	message     string
+	errCallback chan error
+}
+
+func NewUserHandler(conn *SockChatWS, store SockchatChannelStore) *UserHandler {
+	handler := UserHandler{
+		Nick:                  conn.Nick,
+		connections:           make(map[*SockChatWS]bool),
+		createChannelRequests: make(chan *CreateChannel),
+		joinChannelRequests:   make(chan *JoinChannel),
+		leaveChannelRequests:  make(chan *LeaveChannel),
+		sendMessageRequests:   make(chan *SendMessage),
+		channelStore:          store,
+	}
+	go handler.HandleRequests()
+	return &handler
+}
+
+func (u *UserHandler) HandleRequests() {
+	for {
+		select {
+		case req := <-u.createChannelRequests:
+			err := u.channelStore.CreateChannel(req.name)
+			req.errCallback <- err
+
+		case req := <-u.joinChannelRequests:
+			err := u.channelStore.AddUserToChannel(req.channel, u)
+			req.errCallback <- err
+
+		case req := <-u.leaveChannelRequests:
+			err := u.channelStore.RemoveUserFromChannel(req.channel, u)
+			req.errCallback <- err
+
+		case req := <-u.sendMessageRequests:
+			channel, err := u.channelStore.GetChannel(req.channel)
+			if !channel.GetMembers()[u] {
+				req.errCallback <- ErrUserNotInChannel
+				continue
+			}
+			req.errCallback <- err
+			channel.MessageMembers(NewSocketMessage("new_message", MessageEvent{req.message, req.channel, u.Nick}))
+		}
+	}
+}
+
+func (u *UserHandler) RequestNewChannel(c ChannelRequest, store SockchatChannelStore) error {
```

Comment:
```suggestion
func (u *UserHandler) RequestNewChannel(c ChannelRequest) error {
```
Function parameter with `SockChatChannelStore` is not needed anymore, since it is the part of `UserHandler` now